src/main/java/com/es/phoneshop/model/history/DefaultSearchHistoryService.java

```diff
@@ -0,0 +1,33 @@
+package com.es.phoneshop.model.history;
+
+import com.es.phoneshop.model.product.Product;
+import jakarta.servlet.http.HttpServletRequest;
+import jakarta.servlet.http.HttpSession;
+
+import java.util.LinkedList;
+import java.util.List;
+
+public class DefaultSearchHistoryService implements SearchHistoryService {
+
+
+    @Override
+    public void Update(Product product, HttpServletRequest request) {
+        HttpSession session = request.getSession();
+        List<Product> products = (List<Product>) session.getAttribute("recentProducts");
```

Comment:
Please move "recentProducts" to class constants like it's done for cart service 
+ move size of recently viwed products to constants 
+ rename method so it starts with lower case